libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -48,80 +47,87 @@ const getSheetValue = (data, key) => data?.find(
-      ?.map((item) => item?.[KEYS.CONFIG.PROMOTE_IGNORE_PATHS])
-      .join(','),
-  };
+  context.enablePromote = getSheetValue(grayboxData, KEYS.CONFIG.ENABLE_PROMOTE)?.toLowerCase() === 'true';
+  context.promoteUrl = getSheetValue(
+    grayboxData,
+    KEYS.CONFIG.PROMOTE_URL[(grayboxIoEnv || 'prod').toUpperCase()],
+  );
+  context.setup.draftsOnly = getSheetValue(
+    grayboxData,
+    KEYS.CONFIG.PROMOTE_DRAFTS_ONLY,
+  )?.toLowerCase() === 'true';
+  context.setup.promoteIgnorePaths = sheet.promoteignorepaths?.data
+    ?.map((item) => item?.[KEYS.CONFIG.PROMOTE_IGNORE_PATHS])
+    .join(',');
 };
 
-const getSharepointData = async (url) => {
-  const liveOrigin = url.origin.replace('-graybox', '').replace('.hlx.page', '.hlx.live');
+const getSharepointData = async (context) => {
+  const liveOrigin = context.liveUrl.origin.replace('-graybox', '').replace('.hlx.page', '.hlx.live');
   const { sharepoint } = await getServiceConfig(liveOrigin);
-  return {
-    root: `/${sharepoint.rootMapping}`,
-    gbRoot: `/${sharepoint.rootMapping}-graybox`,
-    driveId: sharepoint.driveId,
-  };
+  context.setup.rootFolder = `/${sharepoint.rootMapping}`;
+  context.setup.gbRootFolder = `/${sharepoint.rootMapping}-graybox`;
+  context.setup.driveId = sharepoint.driveId;
 };
 
-const getFilePath = async ({ ref, repo, owner, excelRef } = {}) => {
+const getFilePath = async (context) => {
+  const { ref, repo, owner, excelRef } = context;
   const status = await fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`);
   if (!status.ok) throw new Error('Failed to fetch file path');
-  return (new URL((await status.json())?.preview?.url))?.pathname;
+  const statusResp = await status.json();
+  context.previewUrl = statusResp.preview.status === 200 && new URL(statusResp.preview.url);
+  context.liveUrl = statusResp.live.status === 200 && new URL(statusResp.live.url);
+  context.setup.projectExcelPath = (new URL(statusResp.preview.url)).pathname.replace('.json', '.xlsx');
 };
 
-const getPreviewUrl = async ({ owner, repo, ref, filePath } = {}) => {
-  const res = await fetch(`${ADMIN}/preview/${owner}/${repo}/${ref}${filePath}`, { method: 'POST' });
+const preview = async (context) => {
+  const { owner, repo, ref, setup: { projectExcelPath } } = context;
+  const res = await fetch(`${ADMIN}/preview/${owner}/${repo}/${ref}${projectExcelPath.replace('.xlsx', '.json')}`, { method: 'POST' });
   if (!res.ok) throw new Error('Failed to fetch preview URL');
-  return new URL((await res.json())?.preview?.url);
+  context.previewUrl = new URL((await res.json()).preview.url);
 };
 
+const publish = async (context) => {
+  const { owner, repo, ref, setup: { projectExcelPath } } = context;
+  const res = await fetch(`${ADMIN}/live/${owner}/${repo}/${ref}${projectExcelPath.replace('.xlsx', '.json')}`, { method: 'POST' });
+  if (!res.ok) throw new Error('Failed to fetch live URL');
+  context.liveUrl = new URL((await res.json()).live.url);
```

Comment:
should wrap in a try catch to ensure this doesn't break

Reply:
This is not needed as the error will bubble up and be caught by the lit component task in [here](https://github.com/adobecom/milo/pull/2584/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR186). That would be an advantage of using lit components, no need for "manual" try/catch and handling the display of the error.